Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/shortcuts/AppShortcutsHelper.kt

```diff
@@ -0,0 +1,57 @@
+package de.rki.coronawarnapp.util.shortcuts
+
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.core.content.pm.ShortcutInfoCompat
+import androidx.core.content.pm.ShortcutManagerCompat
+import androidx.core.graphics.drawable.IconCompat
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.ui.launcher.LauncherActivity
+import de.rki.coronawarnapp.util.AppShortcuts
+import de.rki.coronawarnapp.util.di.AppContext
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AppShortcutsHelper @Inject constructor(@AppContext private val context: Context) {
+
+    suspend fun restoreAppShortcut() = withContext(Dispatchers.IO) {
+        if (ShortcutManagerCompat.getDynamicShortcuts(context).size == 0) {
+            val shortcut = ShortcutInfoCompat.Builder(context, CONTACT_DIARY_SHORTCUT_ID)
+                .setShortLabel(context.getString(R.string.app_shortcut_contact_diary_title))
+                .setLongLabel(context.getString(R.string.app_shortcut_contact_diary_title))
+                .setIcon(IconCompat.createWithResource(context, R.drawable.ic_contact_diary_shortcut_icon))
+                .setIntent(createContactDiaryIntent())
+                .build()
+
+            ShortcutManagerCompat.addDynamicShortcuts(context, listOf(shortcut))
+        }
+    }
+
+    fun disableAppShortcut() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
```

Comment:
This check is not needed, ShortcutManagerCompat.removeDynamicShortcuts already checks for it.